includes/class-wc-order-item.php

```diff
@@ -0,0 +1,412 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Order Item
+ *
+ * A class which represents an item within an order and handles CRUD.
+ * Uses ArrayAccess to be BW compatible with WC_Orders::get_items().
+ *
+ * @version     2.7.0
+ * @since       2.7.0
+ * @package     WooCommerce/Classes
+ * @author      WooThemes
+ */
+class WC_Order_Item extends WC_Data implements ArrayAccess {
+
+	/**
+	 * Data array, with defaults.
+	 * @since 2.7.0
+	 * @var array
+	 */
+	protected $_data = array(
+		'order_id'      => 0,
+		'order_item_id' => 0,
+		'name'          => '',
+		'type'          => '',
+	);
+
+	/**
+	 * Stores meta in cache for future reads.
+	 * A group must be set to to enable caching.
+	 * @var string
+	 */
+	protected $_cache_group = 'order_itemmeta';
+
+	/**
+	 * Meta type. This should match up with
+	 * the types avaiable at https://codex.wordpress.org/Function_Reference/add_metadata.
+	 * WP defines 'post', 'user', 'comment', and 'term'.
+	 */
+	protected $_meta_type = 'order_item';
+
+	/**
+	 * Constructor.
+	 * @param int|object|array $order_item ID to load from the DB (optional) or already queried data.
+	 */
+	public function __construct( $item = 0 ) {
+		if ( $item instanceof WC_Order_Item ) {
+			if ( $this->is_type( $item->get_type() ) ) {
+				$this->set_all( $item->get_data() );
+			}
+		} elseif ( is_array( $item ) ) {
+			$this->set_all( $item );
+		} else {
+			$this->read( $item );
+		}
+	}
+
+	/**
+	 * Set all data based on input array.
+	 * @param array $data
+	 * @access private
+	 */
+	public function set_all( $data ) {
+		foreach ( $data as $key => $value ) {
+			if ( is_callable( array( $this, "set_$key" ) ) ) {
+				$this->{"set_$key"}( $value );
+			} else {
+				$this->_data[ $key ] = $value;
+			}
+		}
+	}
+
+	/**
+	 * Type checking
+	 * @param  string|array  $Type
+	 * @return boolean
+	 */
+	public function is_type( $type ) {
+		return is_array( $type ) ? in_array( $this->get_type(), $type ) : $type === $this->get_type();
+	}
+
+	/**
+	 * Get qty.
+	 * @return int
+	 */
+	public function get_qty() {
+		return 1;
+	}
+
+	/*
+	|--------------------------------------------------------------------------
+	| Getters
+	|--------------------------------------------------------------------------
+	*/
+
+	/**
+	 * Get order item ID.
+	 * @return int
+	 */
+	public function get_id() {
+		return $this->get_order_item_id();
+	}
+
+	/**
+	 * Get order ID this meta belongs to.
+	 * @return int
+	 */
+	public function get_order_id() {
+		return absint( $this->_data['order_id'] );
+	}
+
+	/**
+	 * Get order item ID this meta belongs to.
+	 * @return int
+	 */
+	public function get_order_item_id() {
```

Comment:
Makes sense. It's a minor issue, but the alias will still lead to uncertainty. i.e. should a developer use `$order_item->get_id()` or `$order_item->get_order_item_id()`?

When I first came across the `$order->customer_user` and `$order->user_id` properties years ago, I thought maybe they referred to two different things. After I found they referred to the same data, I wasn't sure whether to use `$order->customer_user` or `$order->user_id` because WC core used both interchangeably. This will be similar.